### PR TITLE
Add IP validator for tinkerbell reconciler

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Description of changes:*
Add IP validator for tinkerbell reconciler

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cxbrowne1207 — APPROVED** [review]
/

File: pkg/providers/tinkerbell/reconciler/reconciler.go [hunk 1]
```diff
@@ -4,27 +4,48 @@ import (
...
 
 // Reconcile reconciles cluster to desired state.
 func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
 	// Implement reconcile all here.
-	// This would include IP validator, validating machine and datacenter configs
+	// This would include validating machine and datacenter configs
 	// and reconciling cp and worker nodes.
 
-	return controller.Result{}, nil
+	log = log.WithValues("provider", "tinkerbell")
```
Comment (panktishah26) [inline pkg/providers/tinkerbell/reconciler/reconciler.go:40]: Maybe not in this PR but should we set the provider value from a constant variable for ex https://github.com/aws/eks-anywhere/blob/main/pkg/constants/constants.go#L36 for all the provider?

**Pull request merged**